libs/barista-components/quick-filter/src/state/reducer.ts

```diff
@@ -46,14 +46,14 @@ export function quickFilterReducer(
 
   switch (action.type) {
     case ActionType.SWITCH_DATA_SOURCE:
+      const stateBackup = state;
       if (state.dataSource) {
         state.dataSource.disconnect();
       }
-      return { ...initialState, dataSource: action.payload };
+      return { ...stateBackup, dataSource: action.payload };
```

Comment:
```suggestion
      return { ...state, dataSource: action.payload };
```
You can spread the state directly, no need to store it in a variable :)